homeassistant/components/mqtt/__init__.py

```diff
@@ -46,15 +46,17 @@
 ATTR_TOPIC = 'topic'
 ATTR_PAYLOAD = 'payload'
 ATTR_QOS = 'qos'
+ATTR_RETAIN = 'retain'
 
 MAX_RECONNECT_WAIT = 300  # seconds
 
 
-def publish(hass, topic, payload, qos=None):
+def publish(hass, topic, payload, qos=None, retain=False):
```

Comment:
This should be handled just like QOS so `retain=None` and an `if retain is not None:` check